extension/src/notification/hooks/useNotificationLogic.ts

```diff
@@ -0,0 +1,104 @@
+import { useAtomValue } from "jotai";
+import { useState, useEffect } from "react";
+
+import { readyAtom } from "../../prototypes/view/states/app";
+import {
+  isEnableAtom,
+  contentAtom,
+  startTimeAtom,
+  finishTimeAtom,
+} from "../../shared/states/environmentVariables";
+
+// Function to generate a hash from content
+async function generateContentId(content: string): Promise<string> {
+  const encoder = new TextEncoder();
+  const data = encoder.encode(content);
+  const hashBuffer = await crypto.subtle.digest("SHA-256", data);
+  const hashArray = Array.from(new Uint8Array(hashBuffer));
+  const hashHex = hashArray.map(byte => byte.toString(16).padStart(2, "0")).join("");
+  return hashHex;
+}
+
+export const useNotificationLogic = () => {
+  const ready = useAtomValue(readyAtom);
+  const [visible, setVisible] = useState(true);
+  const [doNotShowAgain, setDoNotShowAgain] = useState<boolean>(false);
+  const [show, setShow] = useState<boolean>(false);
+
+  const isEnable = useAtomValue(isEnableAtom);
+  const content = useAtomValue(contentAtom);
+  const startTime = useAtomValue(startTimeAtom);
+  const finishTime = useAtomValue(finishTimeAtom);
+
+  const [contentId, setContentId] = useState<string | null>(null);
+
+  useEffect(() => {
+    const updateContentId = async () => {
+      if (content) {
+        const newContentId = await generateContentId(content);
+        setContentId(newContentId);
+
+        const storedDoNotShowAgain = localStorage.getItem("doNotShowAgain");
+        const storedContentID = localStorage.getItem("contentId");
```

Comment:
I think we can define the keys as constaints and use them when both get and set.
For the nameing i suggest follow some format that already in use, with prefix `PLATEAUVIEW3_STORAGE_`, for example `PLATEAUVIEW3_STORAGE_NOTIFICATION_DO_NOT_SHOW_AGAIN`

Reply:
sounds good!

Reply:
I added one commit to use constaints for the keys. Thank you!